plutus-core/src/Language/PlutusCore/Evaluation/Machine/ExBudgeting.hs

```diff
@@ -142,30 +142,28 @@ instance ToExMemory (Term TyName Name uni ()) where
 instance ToExMemory (Term TyName Name uni ExMemory) where
     toExMemory = termAnn
 
+-- | A class for injecting a 'BuiltinName' into an @exBudgetCat@.
+-- We need it, because the constant application machinery calls 'spendBudget' before reducing a
```

Comment:
We could pass `BuiltinName -> exBudgetCat` function into the CAM. Since it's only used at the one spot, maybe that would be enough?

Reply:
> We could pass `BuiltinName -> exBudgetCat` function into the CAM. Since it's only used at the one spot, maybe that would be enough?

And write it out it all the way down to `applyTypeSchemed` as an explicit argument? That would be annoying. And if it's not explicit and passed in a `MonadReader` or whatever, then we don't win anything.

Reply:
Fair enough.